### PR TITLE
filter .keymaterial files in listStatus

### PR DESCRIPTION
**Pull request by ellisjoe**
@uschi2000

### CODE REVIEW

File: hadoop-crypto/src/main/java/com/palantir/hadoop/ConfigurableEncryptedFileSystem.java [hunk 1]
```diff
@@ -82,6 +87,19 @@ public URI getUri() {
         return setUriSchemeFunc(encryptedScheme).apply(fs.getUri());
     }
 
+    @Override
+    // TODO(jellis): consider moving logic related to FileKeyStorageStrategy into a separate FileSystem
+    public FileStatus[] listStatus(Path path) throws IOException {
+        Collection<FileStatus> files = Collections2.filter(Arrays.asList(fs.listStatus(path)),
+                new Predicate<FileStatus>() {
```
Comment (uschi2000) [inline hadoop-crypto/src/main/java/com/palantir/hadoop/ConfigurableEncryptedFileSystem.java]: static?

File: hadoop-crypto/src/main/java/com/palantir/hadoop/ConfigurableEncryptedFileSystem.java [hunk 2]
```diff
@@ -82,6 +87,19 @@ public URI getUri() {
         return setUriSchemeFunc(encryptedScheme).apply(fs.getUri());
     }
 
+    @Override
+    // TODO(jellis): consider moving logic related to FileKeyStorageStrategy into a separate FileSystem
+    public FileStatus[] listStatus(Path path) throws IOException {
+        Collection<FileStatus> files = Collections2.filter(Arrays.asList(fs.listStatus(path)),
```
Comment (uschi2000) [inline hadoop-crypto/src/main/java/com/palantir/hadoop/ConfigurableEncryptedFileSystem.java]: hmm, should we avoid going array->list->array?

File: hadoop-crypto/src/test/java/com/palantir/hadoop/ConfigurableEncryptedFileSystemTest.java [hunk 3]
```diff
@@ -167,6 +169,20 @@ public void testFileNameCollidesWithKeyMaterial() throws IOException {
         efs.open(path);
     }
 
+    @Test
+    public void testListStatus_keyMaterialFilesFiltered() throws IOException {
```
Comment (uschi2000) [inline hadoop-crypto/src/test/java/com/palantir/hadoop/ConfigurableEncryptedFileSystemTest.java]: curious why we never hit this bug before?

Comment (ellisjoe) [reply hadoop-crypto/src/test/java/com/palantir/hadoop/ConfigurableEncryptedFileSystemTest.java]: a FileSystem higher up intercepted the listStatus calls and never directed them at the backing store

File: hadoop-crypto/src/main/java/com/palantir/hadoop/ConfigurableEncryptedFileSystem.java (hunk 1 above)
Comment (ellisjoe) [reply hadoop-crypto/src/main/java/com/palantir/hadoop/ConfigurableEncryptedFileSystem.java]: done

File: hadoop-crypto/src/main/java/com/palantir/hadoop/ConfigurableEncryptedFileSystem.java (hunk 2 above)
Comment (ellisjoe) [reply hadoop-crypto/src/main/java/com/palantir/hadoop/ConfigurableEncryptedFileSystem.java]: Arrays.asList doesn't copy so we should be fine. Gonna need array -> array anyway so this should be equivalent.

File: hadoop-crypto/src/test/java/com/palantir/hadoop/ConfigurableEncryptedFileSystemTest.java (hunk 3 above)
Comment (uschi2000) [reply hadoop-crypto/src/test/java/com/palantir/hadoop/ConfigurableEncryptedFileSystemTest.java]: oh, i see

**Pull request merged**